### PR TITLE
Add enhancement for ncp-operator config

### PR DESCRIPTION
**Pull request by jwsui**
Make ncp-operator responsible for inferring adaptor by os-release of host node, and remove coe.adaptor from ConfigMap. Add check for immutable fields to prevent these fields from being changed.

### CODE REVIEW

File: pkg/controller/configmap/config.go [hunk 1]
```diff
@@ -571,6 +578,29 @@ func NeedApplyChange(currConfig *corev1.ConfigMap, prevConfig *corev1.ConfigMap)
 	return needChange, nil
 }
 
+func immutableFieldChanged(cur, prev *ini.File) bool {
+	immutableFields := []struct {
+		Section string
+		Key     string
+	}{
```
Comment (dantingl) [inline pkg/controller/configmap/config.go:585]: Do we need a const dict to save the options?

File: pkg/controller/sharedinfo/shared_info.go [hunk 2]
```diff
@@ -40,33 +40,35 @@ type SharedInfo struct {
 	NsxNcpDeploymentSpec  *unstructured.Unstructured
 }
 
+func getAdaptorName() (string, error) {
+	osReleasePath, envSet := os.LookupEnv("OS_RELEASE_PATH")
+	if !envSet {
+		osReleasePath = "/host/etc/os-release"
```
Comment (dantingl) [inline pkg/controller/sharedinfo/shared_info.go:46]: OsReleaseFile in https://github.com/vmware/nsx-container-plugin-operator/blob/52d9f351cc7812a7c3878833453640638a478df9/pkg/types/names.go#LL41C2-L41C15

Comment (jwsui) [reply pkg/controller/sharedinfo/shared_info.go]: Thanks, done.

File: pkg/controller/configmap/config.go (hunk 1 above)
Comment (jwsui) [reply pkg/controller/configmap/config.go]: Golang does not support constant arrays, maps or slices, and I think we can use global variable instead.

**Review by dantingl — DISMISSED** [review]

**Review by dantingl — APPROVED** [review]

**Pull request merged**